main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -679,6 +703,30 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ "(" ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ ")" ~ SP ~> ParsedAst.Expression.Tuple
     }
 
+    def ArrayLit: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("[|") ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ atomic("|]") ~ SP ~> ParsedAst.Expression.ArrayLit
+    }
+
+    def ArrayNew: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("[|") ~ optWS ~ Expression ~ optWS ~ ";" ~ optWS ~ Expression ~ optWS ~ atomic("|]") ~ SP ~> ParsedAst.Expression.ArrayNew
+    }
+
+    def ArrayLength: Rule1[ParsedAst.Expression] = rule {
+      SP ~ "[" ~ optWS ~ Expression ~ optWS ~ "]" ~ atomic("length") ~ SP ~> ParsedAst.Expression.ArrayLength
```

Comment:
I have a crazy suggestion (which might not be possible -- discuss!)

In math we often write `|x|` for the length. Could we write e.g.:

```
def f(a: Array[Int]): Int = |a|
```

for the length of an array? (Do you foresee any problems?)